vm/devices/storage/disk_nvme/nvme_driver/src/driver.rs

```diff
@@ -53,6 +63,10 @@ pub struct NvmeDriver<T: DeviceBacking> {
     io_issuers: Arc<IoIssuers>,
     #[inspect(skip)]
     rescan_event: Arc<event_listener::Event>,
+    /// NVMe namespace associated with this driver (1:1).
+    #[inspect(skip)]
+    namespace: Option<Arc<Namespace>>,
```

Comment:
No? We support more than one namespace per controller.

Reply:
From host side yes - we provide BDF and namespace ID, and those namespaces can be from same or different controllers.
But looking through OpenHCL code it looks like we always resolve pci_id+nsid into dedicated SimpleDisk, which is separate NvmeDriver object for each namespace.

Reply:
I don't think the resolving is true, as you comment. NvmeDiskResolver::resolve will look up the namespace (and return a 1:1 mapping of NvmeDisk to namespace).

The code that gets this namespace (NvmeManagerWorker::get_namespace) will first get the driver (one per pci_id) and then get the namespace for that driver.

There must be a 1:n mapping of driver : namespaces, or attaching multiple disks backed by the same nvme controller will not work. (And, that works).

Reply:
Looks like you're right. NvmeDriver can return multiple namespaces - each of them will be one SimpleDisk.